lib/embed.dart

```diff
@@ -1292,6 +1370,39 @@ class EmbedContext extends Context {
 
   @override
   bool get isFocused => focusedEditor == 'dart' && editor.hasFocus;
+
+  // Character constants.
+  static const int _lf = 10;
+  static const int _cr = 13;
+
+  /// Count lines in string (CR, LF or CR-LF can be line separators).
+  /// (Gives correct result as opposed to .split('\n').length, which
+  /// reports 1 extra line in many cases, and this does so without the
+  /// extra work .split() would do by creating the list of copied strings).
+  int countLinesInString(String str) {
+    final List<int> data = str.codeUnits;
+    int lines = 0;
+    final int end = data.length;
+    int sliceStart = 0;
+    int char = 0;
+    for (int i = 0; i < end; i++) {
+      final int previousChar = char;
+      char = data[i];
+      if (char != _cr) {
+        if (char != _lf) continue;
+        if (previousChar == _cr) {
+          sliceStart = i + 1;
+          continue;
+        }
+      }
+      lines++;
+      sliceStart = i + 1;
+    }
+    if (sliceStart < end) {
+      lines++;
+    }
+    return lines;
+  }
```

Comment:
If you `import 'dart:convert';` you can do:

```suggestion
  /// Counts the number of lines in [str].
  static int countLinesInString(String str) =>
      LineSplitter().convert(str).length;
```

Reply:
This countLinesInString() essentially *is* the code from LineSplitter().convert(), minus the memory operations (allocations and copying for all of the lines).  My thought was to eliminate needless memory thrashing just to get a line count.
I wrote the routine originally for dart-services where it is used repeatedly, and it is probably more beneficial to eliminate the needless memory thrash, but I thought I would use it here also.
Would you like me to replace it anyway ?

Reply:
I still think I would prefer to use the library implementation here, even if we risk causing memory issues. Are you seeing memory issues with `LineSplitter`?

Reply:
I have changed this to use LineSplitter().convert(str).length to count lines.
I did not see any memory allocation issues with LineSplitter, per se, and in this instance we don't call *too* often and this is on the user's machine anyway.